### PR TITLE
Early Refactor Attempt at Some Small Stuff Here and There...

### PR DESCRIPTION
**Pull request by vchadha**
What a PR name...

tl;dr

- Fix up plant code to be more generic
- Watering can can be hidden if we want
- Delivery manager code is cleaned up a bit
- UI code is in UI controller now
- Removed unnecessary scenes and improved menu logic
- Allow for arbitrary number of digits in score
- Make sound controller a bit more generic
- Some other small stuff

### CODE REVIEW

File: Assets/Scripts/GameObjects/Peppers/PepperPlant.cs [hunk 1]
```diff
@@ -2,94 +2,126 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+// TODO: add option to disable wateringcan
 public class PepperPlant : InteractableObject
 {
     public override bool isInteractable { get { return true; }}
+
+    // TODO: i feel like most of this should be empty list because this creates an entity to drop on a pizza, right? It is not, itself, able to interact with the pizza.
```
Comment (justinro-underscore) [inline Assets/Scripts/GameObjects/Peppers/PepperPlant.cs:10]: But isn't the droppable object that is spawned the pepper? So technically it's being controlled by the PepperPlant? I'm not sure

File: Assets/Scripts/Utils/Constants.cs [hunk 2]
```diff
@@ -55,4 +55,6 @@ public enum Pineapple
     }
 
     public const int maxIngredients = 10;
+
+    public const int MAX_MEAT_DELIVERY_SPOTS = 4;
```
Comment (justinro-underscore) [inline Assets/Scripts/Utils/Constants.cs:59]: Love me some constants

File: Assets/Scripts/Utils/Utilities.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
+
+public class Utilities
+{
+    // Return number of digits in an int
+    public static int countDigitsInt( int num )
```
Comment (justinro-underscore) [inline Assets/Scripts/Utils/Utilities.cs:6]: Huge nit: could replace this function with
```
(int)Math.Log(num, 10) + 1
```

**Review by justinro-underscore — APPROVED** [review]

File: Assets/Scripts/GameObjects/Peppers/PepperPlant.cs (hunk 1 above)
Comment (vchadha) [reply Assets/Scripts/GameObjects/Peppers/PepperPlant.cs:10]: honestly good question haha

File: Assets/Scripts/Utils/Utilities.cs (hunk 3 above)
Comment (vchadha) [reply Assets/Scripts/Utils/Utilities.cs:6]: You beautiful man
Next version 😆

**Pull request merged**